core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -1695,6 +1766,118 @@ public void testSeqNoAndCheckpoints() throws IOException {
         }
     }
 
+    // this test writes documents to the engine while concurrently flushing/commit
+    // and ensuring that the commit points contain the correct sequence number data
+    public void testConcurrentWritesAndCommits() throws Exception {
+        try (final Store store = createStore();
+             final InternalEngine engine = new InternalEngine(config(defaultSettings, store, createTempDir(), newMergePolicy(),
+                                                                     new SnapshotDeletionPolicy(NoDeletionPolicy.INSTANCE),
+                                                                     IndexRequest.UNSET_AUTO_GENERATED_TIMESTAMP, null))) {
+
+            final int numIndexingThreads = randomIntBetween(4, 7);
+            final int numDocsPerThread = randomIntBetween(500, 1000);
```

Comment:
Average run is about 2.2 seconds on my Mac.  I agree, could be lighter.. but dropping to `randomIntBetween(100,500)` reduces the execution time to 1.95 secs on average.  Not sure if its worth going lower than that as we likely won't often get useful commits to check against (all docs would go in a single commit).